### PR TITLE
Add relation dialog improvements

### PR DESCRIPTION
**Pull request by nyalldawson**
- Swaps the dialog so parent layer is defined first. I *ALWAYS* get this backwards, and I think it's because the dialog is backwards and it's more natural to define the parent first
- Use proper layer/field comboboxes in dialog

### CODE REVIEW

File: src/app/qgsrelationadddlg.cpp [hunk 1]
```diff
@@ -24,45 +25,41 @@ QgsRelationAddDlg::QgsRelationAddDlg( QWidget *parent )
 {
   setupUi( this );
 
-  mTxtRelationId->setPlaceholderText( tr( "[Generated automatically]" ) );
-  checkDefinitionValid();
+  connect( mCbxReferencingLayer, &QgsMapLayerComboBox::layerChanged, this, [=]( QgsMapLayer * layer ) { mCbxReferencingField->setLayer( layer ); }
```
Comment (nyalldawson) [inline src/app/qgsrelationadddlg.cpp]: `QgsMapLayer* layer`

Everything else is just plain evil. I just can't get qt creator to reliably respect this setting. I wonder if astyle could handle this...

Comment (nyalldawson) [inline src/app/qgsrelationadddlg.cpp]: Seems not supported in the embedded astyle version, but supported in more recent versions. How's #3463 going?

**Comment by nyalldawson** [comment]
@m-kuhn so are you in favour of this?

**Comment by m-kuhn** [comment]
Yes, totally

**Comment by nyalldawson** [comment]
@m-kuhn thanks! I'll address the pointer stuff after the astyle update lands

**Pull request merged**